src/classes/AllocationParent.cls

```diff
@@ -0,0 +1,131 @@
+public inherited sharing class AllocationParent {
+    private Id id;
+    private SObjectType objectType = null;
+    private SObject record;
+    private Decimal donationAmount = 0;
+    private String currencyCode;
+    private String keyPrefix;
+    private String objectApiName;
+    private String objectLabelPlural;
+
+    private final String OPPORTUNITY_AMOUNT = String.valueOf(Opportunity.SObjectType.fields.Amount);
+    private final String PAYMENT_AMOUNT = String.valueOf(npe01__OppPayment__c.SObjectType.fields.npe01__Payment_Amount__c);
+
+    public AllocationParent() {}
+
+    public AllocationParent(Id parentId) {
+        id = parentId;
+        objectType = id.getSObjectType();
+        record = getRecord(objectType);
+        setPropertiesFromDescribe();
+
+        if (UserInfo.isMultiCurrencyOrganization()) {
+            currencyCode = (String)record.get(UTIL_Currency.CURRENCY_ISO_CODE_FIELD);
+        }
+    }
+
+    public Id getId() {
+        return id;
+    }
+
+    public SObjectType getSObjectType() {
+        return objectType;
+    }
+
+    public String getKeyPrefix() {
+        return keyPrefix;
+    }
+
+    public String getObjectLabelPlural() {
+        return objectLabelPlural;
+    }
+
+    public String getObjectApiName() {
+        return objectApiName;
+    }
+
+    public String getRecordName() {
+        return (String)record?.get('Name');
+    }
+
+    public String getCurrencyCode() {
+        return currencyCode;
+    }
+
+    public String getAllocationLookupApiName() {
+        return allocationLookupApiNameBySObjectType.get(objectType);
+    }
+
+    public Boolean isAnOpportunity() {
+        return objectType == Opportunity.SObjectType;
+    }
+
+    public Boolean isAPayment() {
```

Comment:
I'm not too keen on the articles and since we generally haven't used them I'll go ahead update these changes to conform to what we've historically done 👍